include/deal.II/base/mpi_noncontiguous_partitioner.h

```diff
@@ -79,6 +79,14 @@ namespace Utilities
        * Fill the vector @p ghost_array according to the precomputed communication
        * pattern with values from @p locally_owned_array.
        *
+       * In the default case, only one object is communicated per entry
+       * (`n_components_templated == 1'). If you want to communicate more
+       * entries, you can increase the value of @p n_components_templated in the
+       * case that you know the size at compile time. If you want to set the
+       * size during runtime, you can set @p n_components. However,
+       * @p n_components_templated has to be set to
+       * `numbers::invalid_unsigned_int` in this case.
```

Comment:
So this means that the compile-time variant via the template takes precedence over the run-time variable passed as an argument to the function?

Reply:
As a follow-up question, is `invalid_unsigned_int` the most intuitive setting for the run-time bounds? I would have intuitively picked `n_components_templated = 0` to switch to the run-time bounds, because I would not see what zero components would mean otherwise, but I think either one is fine.

Reply:
> So this means that the compile-time variant via the template takes precedence over the run-time variable passed as an argument to the function?

Only one can be set. It is checked via an assert. I have added a comment to the documentation.

> As a follow-up question, is invalid_unsigned_int the most intuitive setting for the run-time bounds? I would have intuitively picked n_components_templated = 0

I have switched from -1 to 0.